homeassistant/components/osoenergy/water_heater.py

```diff
@@ -0,0 +1,304 @@
+            vol.Required(ATTR_PROFILE_HOURS["11"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["12"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["13"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["14"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["15"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["16"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["17"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["18"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["19"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["20"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["21"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["22"]): vol.Coerce(int),
+            vol.Required(ATTR_PROFILE_HOURS["23"]): vol.Coerce(int),
+        },
+        "async_set_profile",
+    )
+
+
+def _get_utc_hour(local_hour: int):
+    """Get the utc hour."""
+    now = dt_util.now()
+    local_time = now.replace(hour=local_hour, minute=0, second=0, microsecond=0)
+    utc_hour = dt_util.as_utc(local_time)
+    return utc_hour.hour
+
+
+def _get_local_hour(utc_hour: int):
+    """Get the utc hour."""
+    now = dt_util.utcnow()
+    now_local = dt_util.now()
+    utc_time = now.replace(hour=utc_hour, minute=0, second=0, microsecond=0)
+    local_hour = dt_util.as_local(utc_time)
+    local_hour = local_hour.replace(day=now_local.day)
+    return local_hour
+
+
+def _convert_profile_to_local(values):
+    """Convert UTC profile to local."""
+    profile = [None] * 24
+    for hour in range(24):
+        local_hour = _get_local_hour(hour)
+        local_hour_string = local_hour.strftime("%Y-%m-%dT%H:%M:%S%z")
+        profile[local_hour.hour] = {local_hour_string: values[hour]}
+
+    return profile
+
+
+class OSOEnergyWaterHeater(OSOEnergyEntity, WaterHeaterEntity):
+    """OSO Energy Water Heater Device."""
+
+    _attr_operation_list = OPERATION_LIST
+    _attr_supported_features = SUPPORT_FLAGS_HEATER
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    @property
+    def unique_id(self):
+        """Return unique ID of entity."""
+        return self._unique_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.device["device_id"])},
+            manufacturer=MANUFACTURER,
+            model=self.device["device_type"],
+            name=self.device["device_name"],
+        )
+
+    @property
+    def name(self):
+        """Return the name of the water heater."""
+        return self.device["device_name"]
+
+    @property
+    def available(self):
+        """Return if the device is available."""
+        return self.device.get("attributes", {}).get("available", False)
+
+    @property
+    def temperature_unit(self):
```

Comment:
Can be removed as you have already set it as an attribute above.

Reply:
Removed the temperature_unit